chord_metadata_service/chord/workflows/wdls/experiments_json_with_files.wdl

```diff
@@ -0,0 +1,285 @@
+            if filename in files:
+                file_found = True
+                file_path = os.path.join(root, filename)
+                path_list.append({'filename': filename, 'url': file_path})
+                break
+        if not file_found:
+            print(f'File not found for {filename}')
+
+with open('path_list.json', 'w') as file:
+    json.dump(path_list, file, indent=4)
+    "
+    >>>
+    output {
+        File path_list = "path_list.json"
+    }
+}
+
+task prepare_for_drs {
+    input {
+        File json_path_list
+    }
+
+    command <<<
+    python3 -c "
+import json
+
+with open('~{json_path_list}', 'r') as file:
+    path_list = json.load(file)
+
+consolidated_paths = [str(path['url']).strip() for path in path_list if str(path['url']).strip()]
+
+print(json.dumps(consolidated_paths))
+
+    "
+    >>>
+    output {
+        Array[String] consolidated_paths_for_drs = read_json(stdout())
+    }
+}
+
+task post_to_drs {
+    input {
+        File file_path
+        String drs_url
+        String project_dataset
+        String token
+        Boolean validate_ssl
+    }
+    command <<<
+        project_id=$(python3 -c 'print("~{project_dataset}".split(":")[0])')
+        dataset_id=$(python3 -c 'print("~{project_dataset}".split(":")[1])')
+        curl ~{true="" false="-k" validate_ssl} \
+            -X POST \
+            -F "file=@~{file_path}" \
+            -F "project_id=$project_id" \
+            -F "dataset_id=$dataset_id" \
+            -H "Authorization: Bearer ~{token}" \
+            --fail-with-body \
+            "~{drs_url}/ingest"
+    >>>
+    output {
+        String response_message = read_string(stdout())
+    }
+}
+
+task write_drs_responses_to_file {
+    input {
+        Array[String] drs_responses
+    }
+
+    command <<<
+        for response in "~{sep=' ' drs_responses}"; do
+            echo "$response"
+        done > results_post_drs.txt
+    >>>
+
+    output {
+        File results_post_drs = "results_post_drs.txt"
+    }
+}
+
+task parse_txt {
+    input {
+        File txt_responses
+    }
+
+    command <<<
+    python3 -c "
+import json
+
+def parse_drs_response(file_path):
+    with open(file_path, 'r') as file:
+        data = file.read().strip().replace('\n', '').split('} {')
+
+    new_array = []
+    for line in data:
+        information = {'name': '', 'self_uri': ''}
+
+        line = line.strip('{} ').replace('\'', '\"')
+        parts = line.split(', ')
+        for part in parts:
+            key_value = part.split(':', 1)
+            if len(key_value) == 2:
+                key, value = key_value[0].strip(), key_value[1].strip().strip('\"')
+                if 'name' in key:
+                    information['name'] = value
+                elif 'self_uri' in key:
+                    information['self_uri'] = value
+        new_array.append(information)
+
+    with open('processed_drs_responses.json', 'w') as outfile:
+        json.dump(new_array, outfile, indent=4)
```

Comment:
As discussed, we should not parse JSON formated data as simple strings, it is better to load the data in a dict and use the dict methods to extract the relevant information. Otherwise this code will be prone to errors and difficult to debug.
The file at `file_path` sould be a JSON array file, then this task can map each element to a `{'name': 'some name', 'self_uri': 'some uri'}`